newIDE/app/src/EventsSheet/ParameterFields/AnyVariableField.js

```diff
@@ -59,6 +59,21 @@ export default React.forwardRef<ParameterFieldProps, ParameterFieldInterface>(
       [projectScopedContainersAccessor]
     );
 
+    const onVariableEditorApply = React.useCallback(
+      (selectedVariableName: string | null) => {
+        if (
+          selectedVariableName &&
+          selectedVariableName.startsWith(props.value)
+        ) {
+          props.onChange(selectedVariableName);
+        }
+        setEditorOpen(false);
+        if (onInstructionTypeChanged) onInstructionTypeChanged();
```

Comment:
What does this does exactly?

Reply:
I did see that it was ending up in a `forceUpdate`, but I was wondering if it was really a "onInstructionTypeChanged" intent that was conveyed but this, or if this function was "hijacked" to make a forceUpdate happens, despite being not really related to what changed.
So if you think the naming is correct, we can keep it - otherwise we can consider adding another callback even if it does the same thing at the end.

Reply:
ok, I see, It's because the dialog may have done a refactoring on the type of the selected variable. It probably needs a comment.

Reply:
In this case yeah, it's a good idea to tell "// The function MAY have done a change of the type of the selected variable."